### PR TITLE
feat: metrics ui component

### LINKED ISSUE #2747: Dark mode has some irregularities with white borders
### Preliminary Checks

- [X] I have reviewed the documentation: https://unkey.com/docs

- [X] I have searched for existing issues: https://github.com/unkeyed/unkey/issues

- [X] This issue is not a question, general help request, or anything other than a bug report directly related to Unkey. Please ask questions in our Discord community: https://unkey.com/discord.

### Reproduction / Replay Link (Optional)

_No response_

### Issue Summary

Arc 2024-12-12 18 20 00
There are some places where the border got changed to white

### Steps to Reproduce

1. Choose dark mode 
2. create a role / permission or look at the ratelimit metrics

### Expected behavior

To not have the white but old gray border

### Other information

_No response_

### Screenshots

_No response_

### Version info

```markdown
- OS: Macos
- Browser: Latest ARC
```

### PR DESCRIPTION
**Pull request by Flo4604**
## What does this PR do?

Fixes the white metric seperator things
CleanShot 2024-12-12 at 18 19 56@2x

Also changes require to import because the build doesn't work locally for me otherwise.

Fixes #2747

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

Check if any place with a metric component is white in darkmode

### Required

- [X] Filled out the "How to test" section in this PR
- [X] Read [Contributing Guide](./CONTRIBUTING.md)
- [X] Self-reviewed my own code
- [X] Commented on my code in hard-to-understand areas
- [X] Ran `pnpm build`
- [X] Ran `pnpm fmt`
- [X] Checked for warnings, there are none
- [X] Removed all `console.logs`
- [X] Merged the latest changes from main onto my branch with `git pull origin main`
- [X] My changes don't cause any responsiveness issues

### Appreciated

- [X] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Unkey Docs if changes were necessary

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `Metric` component for enhanced display of various metrics across the dashboard.
	- Updated multiple pages to utilize the new `Metric` component for presenting API key metrics, API usage statistics, rate limit information, and more.

- **Bug Fixes**
	- Improved visual styling of alerts and dialog components by adding border styles.

- **Refactor**
	- Removed local definitions of the `Metric` component in favor of a centralized version, streamlining code and enhancing maintainability.

- **Style**
	- Enhanced styling for various components by adding border classes to improve visual consistency.

### CODE REVIEW

**Comment by Flo4604** [comment]
auth page still has another metric component

**Comment by ogzhanolguncu** [comment]
Could you please run the formatter?
`Ran pnpm fmt` you checked this, but code isn't formatted properly.

**Comment by Flo4604** [comment]
> Could you please run the formatter? `Ran pnpm fmt` you checked this, but code isn't formatted properly.

CleanShot 2024-12-12 at 19 01 18@2x
I did, shows no changes for me

File: apps/dashboard/components/ui/metric.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import * as React from "react";
+
+interface MetricProps {
+  label: string;
+  value: string | React.ReactNode;
+}
+
+const Metric = React.forwardRef<HTMLDivElement, MetricProps>(({ label, value }) => {
```
Comment (ogzhanolguncu) [inline apps/dashboard/components/ui/metric.tsx:8]: `forwardRef` looks redundant here.

**Comment by ogzhanolguncu** [comment]
> > Could you please run the formatter? `Ran pnpm fmt` you checked this, but code isn't formatted properly.
> 
> CleanShot 2024-12-12 at 19 01 18@2x I did, shows no changes for me.
> 
> I fixed the stuff manually and it didn't reformat it idk

image

Might be my bad that one caught me off guard. Also can you open an issue for this one?

**Comment by Flo4604** [comment]
> > > Could you please run the formatter? `Ran pnpm fmt` you checked this, but code isn't formatted properly.
> > 
> > 
> > CleanShot 2024-12-12 at 19 01 18@2x I did, shows no changes for me.
> > I fixed the stuff manually and it didn't reformat it idk
> 
> image
> 
> Might be my bad that one caught me off guard. Also can you open an issue for this one?

actually It zed that formatted it like this but biome doesn't seem to care, I manually changed it back and it didn't reformat it so I'm not sure whats up with that.

**Comment by Flo4604** [comment]
should be most if not all places

**Review by ogzhanolguncu — APPROVED** [review]

**Comment by ogzhanolguncu** [comment]
Wow that's a lot of Metric components 😄. Great work.

**Pull request merged**